Providers/SwissBitcoinPayProvider.cs

```diff
@@ -0,0 +1,122 @@
+            ILocalizationService localizationService,
+            ICustomerService customerService,
+            ISettingFactory settingFactory,
+            ICurrencyService currencyService,
+            IOrderService orderService)
+        {
+            _localizationService = localizationService;
+            _services = services;
+            _currencyService = currencyService;
+            _customerService = customerService;
+            _settingFactory = settingFactory;
+            _orderService = orderService;
+        }
+
+        public ILogger Logger { get; set; } = NullLogger.Instance;
+
+        public static string SystemName => "SmartStore.SwissBitcoinPay";
+        public override bool SupportCapture => false;
+        public override bool SupportPartiallyRefund => false;
+        public override bool SupportRefund => false;
+        public override bool SupportVoid => false;
+        public override bool RequiresInteraction => false;
+        public override PaymentMethodType PaymentMethodType => PaymentMethodType.Redirection;
+
+        public RouteInfo GetConfigurationRoute()
+            => new(nameof(SwissBitcoinPayAdminController.Configure), "SwissBitcoinPay", new { area = "Admin" });
+
+
+        public override Widget GetPaymentInfoWidget()
+            => new ComponentWidget(typeof(SwissBitcoinPayViewComponent));
+
+        public override async Task<(decimal FixedFeeOrPercentage, bool UsePercentage)> GetPaymentFeeInfoAsync(ShoppingCart cart)
+        {
+            var settings = await _settingFactory.LoadSettingsAsync<SwissBitcoinPaySettings>(_services.StoreContext.CurrentStore.Id);
+            return (settings.AdditionalFee, settings.AdditionalFeePercentage);
+        }
+
+        public override async Task<ProcessPaymentResult> ProcessPaymentAsync(ProcessPaymentRequest processPaymentRequest)
+        {
+            var result = new ProcessPaymentResult();
+            result.NewPaymentStatus = PaymentStatus.Pending;
+
+            try
+            {
+
+                var myStore = _services.StoreContext.CurrentStore;
+                var settings = await _settingFactory.LoadSettingsAsync<SwissBitcoinPaySettings>(myStore.Id);
+                var myCustomer = await _customerService.GetAuthenticatedCustomerAsync();
+
+                var apiService = new SwissBitcoinPayService();
+                result.AuthorizationTransactionResult = apiService.CreateInvoice(settings, new PaymentDataModel()
+                {
+                    CurrencyCode = _currencyService.PrimaryCurrency.CurrencyCode,
+                    Amount = processPaymentRequest.OrderTotal,
+                    BuyerEmail = "" + myCustomer.Email,
+                    BuyerName = myCustomer.FirstName + " " + myCustomer.LastName,
+                    OrderID = processPaymentRequest.OrderGuid.ToString(),
+                    StoreID = myStore.Id,
+                    CustomerID = myCustomer.Id,
+                    Description = "From " + myStore.Name,

```

Comment:
For easier reading for the shop owner in their dashboard, it's better to include a (short) order ID, like order number.

Reply:
The order ID in Smartstore is a GUID and is not short :)
I've added the customer email.